Refactored expired search and session deletion to delete records in s…

EreMaijala opened the pull request:
…mall batches to avoid locking the tables for too long or creating a massive transaction in case the table has accumulated records for a longer time.

I also added timestamps to the messages the expiration process prints out so that e.g. when there's a gap in the ID range, it doesn't look like nothing's happening. And having the timestamp is useful too when piping the output to a log file.

demiankatz commented:
Thanks for this. I've just taken a very quick glance at this and will probably not have time to do detailed work on it until after vacation... but a few first thoughts to perhaps keep things moving along:

1.) The delete logic is present not just in the console utilities but also in the web-based admin panel (see [this code](https://github.com/vufind-org/vufind/blob/7b4646fed21edbe80ad1a6c9ba5783cbfe28e35b/module/VuFindAdmin/src/VuFindAdmin/Controller/MaintenanceController.php#L130)). Your refactoring to remove getExpiredQuery is going to break the generic design the admin panel uses for expiring various types of things. We either need to keep the old method for backward compatibility or else revise the admin panel code.

2.) It might make sense to add a switch to the command line to set the sleep time for the loop; some people may want this, others may find it problematic. Chunk size might also be a useful configurable setting.

3.) Minor nitpick, but I'd rename the message() method you've added to the console controller to something like datedMessage() to clarify its purpose.

EreMaijala commented:
1.) Oops. I reverted the deletion since I'm not sure the batch stuff works well with the admin panel (which I don't have experience with).

2.) True, done. I modified the help message to accommodate this. Do you think this makes sense? I also simplified the help param check since GetOpt is supposed to always return the non-alias version of the parameter.

3.) Done.

demiankatz commented:
Looks good. Thanks for the revisions! I just made one more small revision to avoid sleeping after the last batch; no point in waiting around once the job is done.
